xoinvader/tests/test_background.py

```diff
@@ -0,0 +1,143 @@
+import pytest
+from copy import copy
+
+from xoinvader.background import Background, Chunk, load_chunks
+from xoinvader.common import Settings
+from xoinvader.utils import Point, Surface
+
+
+PREFIX="xoinvader/tests/fixtures/"
+
+def test_chunk():
+    c = Chunk("test")
+    assert not c.lines
+    assert not c.length
+    assert c.name == "test"
+
+    c.add_line("wut")
+    assert c.lines[0] == "wut"
+    assert c.length == 1
+
+    c[0] = "ugh"
+    assert c[0] == "ugh"
+
+
+def test_load_chunks():
+    with pytest.raises(ValueError):
+        load_chunks(PREFIX + "chunk_no_name.bg")
+
+    with pytest.raises(ValueError):
+        load_chunks(PREFIX + "chunk_duplicate_names.bg")
+
+    with pytest.raises(ValueError):
+        load_chunks(PREFIX + "no_chunks.bg")
+
+    c = load_chunks(PREFIX + "chunk_normal.bg")
+    assert len(c) == 3
+    assert c[0][0] == "qweqwe"
+    assert c[0].length == 1
+
+    d = load_chunks(PREFIX + "chunk_normal.bg", 3)
+    assert d[0][0] == "qwe"
+
+
+def test_background():
+    Settings.layout.field.edge.x = 3
```

Comment:
No extra logic with _w and _h found. You are checking here that python can assign to variable)

```python
assert b._w == Settings.layout.field.edge.x
assert b._h == Settings.layout.field.edge.y
```